gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -1940,6 +1945,50 @@ void GazeboRosApiPlugin::publishSimTime()
   pub_clock_.publish(ros_time_);
 }
 
+void GazeboRosApiPlugin::publishRealTimeFactor(ConstWorldStatisticsPtr &_msg)
```

Comment:
This could just be moved to publishSimTime, ensuring that the /sim_real_time_factor also publishes at the specified pub_clock_frequency

Reply:
I thought about that, but I wanted to keep RTF and sim time logically distinct.  What I've done instead is eliminated the `rtf_sub_` subscriber, and refactored this callback to be triggered on the world updated event (similar to `publishSimTime`)